compiler/AST/AstPrintDocs.cpp

```diff
@@ -98,16 +112,27 @@ bool AstPrintDocs::enterModSym(ModuleSymbol* node) {
   }
 
   // If this is a sub module (i.e. other modules were entered and not yet
-  // exited before this one), ensure the docs naming is correct.
-  if (!this->moduleNames.empty()) {
-    node->addPrefixToName(this->moduleNames.top() + ".");
+  // exited before this one), we want to open a new file in a subdirectory.
+  if (strcmp(node->name, this->moduleName.c_str()) != 0) {
+    // Create a directory with our module name and store this file in it.
+    static const int dirPerms = S_IRWXU | S_IRWXG | S_IRWXO;
+    int result = mkdir(this->pathWithoutPostfix.c_str(), dirPerms);
+    if (result != 0 && errno != 0 && errno != EEXIST) {
+      USR_FATAL(astr("Failed to create directory: ", this->pathWithoutPostfix.c_str(),
+                   " due to: ", strerror(errno)));
+    }
```

Comment:
Here and in docs.cpp you use the mkdir system call. I think it would be better to merge these codepaths with ensureDirExists from files.cpp. You could replace the body of ensureDirExists with this mkdir error-checking snippet (it forks/execs mkdir -p now). I wouldn't try to do that in this PR though.